standalone/test/unit_tests/unittest_udp_client.cpp

```diff
@@ -45,25 +47,61 @@ TEST(UdpClientTests, testInvalidNewDataHandler)
+                   nullptr,
+                   std::bind(&CallbackHandler::handleError, &handler, _1),
+                   HOST_UDP_READ_PORT,
+                   boost::asio::ip::make_address_v4(UDP_MOCK_IP_ADDRESS.c_str()).to_uint(),
+                   UDP_MOCK_SEND_PORT),
+               std::invalid_argument);
+#elif BOOST_VERSION >= 106900
+  EXPECT_THROW(psen_scan_v2_standalone::UdpClientImpl reader(
+                   nullptr,
+                   std::bind(&CallbackHandler::handleError, &handler, _1),
+                   HOST_UDP_READ_PORT,
+                   boost::asio::ip::address_v4::make_address_v4(UDP_MOCK_IP_ADDRESS.c_str()).to_uint(),
+                   UDP_MOCK_SEND_PORT),
                std::invalid_argument);
+#else
+  EXPECT_THROW(psen_scan_v2_standalone::UdpClientImpl reader(
+                   nullptr,
+                   std::bind(&CallbackHandler::handleError, &handler, _1),
+                   HOST_UDP_READ_PORT,
+                   boost::asio::ip::address_v4::from_string(UDP_MOCK_IP_ADDRESS.c_str()).to_ulong(),
+                   UDP_MOCK_SEND_PORT),
+               std::invalid_argument);
+#endif
 }
 
 TEST(UdpClientTests, testInvalidErrorHandler)
 {
   CallbackHandler handler;
-
-  EXPECT_THROW(
-      psen_scan_v2_standalone::UdpClientImpl reader(std::bind(&CallbackHandler::handleNewData, &handler, _1, _2),
-                                                    nullptr,
-                                                    HOST_UDP_READ_PORT,
-                                                    inet_network(UDP_MOCK_IP_ADDRESS.c_str()),
-                                                    UDP_MOCK_SEND_PORT),
-      std::invalid_argument);
+#if BOOST_VERSION > 107000
+  EXPECT_THROW(psen_scan_v2_standalone::UdpClientImpl reader(
+                   std::bind(&CallbackHandler::handleNewData, &handler, _1, _2),
+                   nullptr,
+                   HOST_UDP_READ_PORT,
+                   boost::asio::ip::make_address_v4(UDP_MOCK_IP_ADDRESS.c_str()).to_uint(),
+                   UDP_MOCK_SEND_PORT),
+               std::invalid_argument);
+#elif BOOST_VERSION >= 106900
+  EXPECT_THROW(psen_scan_v2_standalone::UdpClientImpl reader(
+                   std::bind(&CallbackHandler::handleNewData, &handler, _1, _2),
+                   nullptr,
+                   HOST_UDP_READ_PORT,
+                   boost::asio::ip::address_v4::make_address_v4(UDP_MOCK_IP_ADDRESS.c_str()).to_uint(),
+                   UDP_MOCK_SEND_PORT),
+               std::invalid_argument);
+#else
+  EXPECT_THROW(psen_scan_v2_standalone::UdpClientImpl reader(
+                   std::bind(&CallbackHandler::handleNewData, &handler, _1, _2),
+                   nullptr,
+                   HOST_UDP_READ_PORT,
+                   boost::asio::ip::address_v4::from_string(UDP_MOCK_IP_ADDRESS.c_str()).to_ulong(),
+                   UDP_MOCK_SEND_PORT),
+               std::invalid_argument);
+#endif
```

Comment:
Why do we have two variants for converting a string ip into a number:
* `inet_addr()` / `inet_network()`, see: https://github.com/PilzDE/psen_scan_v2/blob/3d7b34894b05b2e2cfb6217321861a4c26b77e5e/standalone/include/psen_scan_v2_standalone/scanner_config_builder.h#L77
* `boost::asio::ip` ... (used 5 times in the tests)

What is the intention behind this?

Reply:
Hi @martiniil I am aware of this. Eventually I think the best way is to just use boost all over. However for now I want to focus on having the tests running on windows without introducing even more Win/Linux defines (by using even more BOOST_VERSION defines ;-) )